assets/js/base/components/text-input/validated-text-input.tsx

```diff
@@ -17,6 +16,29 @@ import { withInstanceId } from '@woocommerce/base-hocs/with-instance-id';
 import TextInput from './text-input';
 import './style.scss';
 
+interface ValidatedTextInputPropsWithId {
+	instanceId?: string;
+	id: string;
+}
+
+interface ValidatedTextInputPropsWithInstanceId {
+	instanceId: string;
+	id?: string;
+}
+
```

Comment:
Essentially this means: if `id` is passed, `instanceId` is optional, and vice/versa.